pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -220,6 +219,28 @@ type PipelineTask struct {
 	Timeout *metav1.Duration `json:"timeout,omitempty"`
 }
 
+// Matrix is used to fan out Tasks in a Pipeline
+type Matrix struct {
+	// Params is a list of parameters used to fan out the pipelineTask
+	// Params takes only `Parameters` of type `"array"`
+	// Each array element is supplied to the `PipelineTask` by substituting `params` of type `"string"` in the underlying `Task`.
+	// The names of the `params` in the `Matrix` must match the names of the `params` in the underlying `Task` that they will be substituting.
+	Params []Param `json:"params,omitempty"`
```

Comment:
you might need a `// +listType=atomic` annotation for these new fields to resolve the build test failures.

Reply:
thanks @lbernick ,I've added the annotation `// +listType=atomic` for `params` filed